syntax = "proto3";

message CreateAccountWeb {
  int64 request_id = 1;
  string email = 2;
  string plaintext_password = 3;
}

message CreateAccountWebSuccess {
  int64 request_id = 1;
  int64 user_id = 2;
}

message CreateAccountWebFailure {
  int64 request_id = 1;
  enum Reason {
    ACCOUNT_ALREADY_EXISTS = 0;
    INVALID_PASSWORD = 1;
    INTERNAL_SERVER_ERROR = 2;
  }
  Reason reason = 2;
}

message AuthenticateWeb {
  int64 request_id = 1;
  string email = 2;
  string plaintext_password = 3;
}

message AuthenticateWebSuccess {
  int64 request_id = 1;
  int64 user_id = 2;
  string creation_date = 3;
}

message AuthenticateWebFailure {
  int64 request_id = 1;
  enum Reason {
    INVALID_CREDENTIALS = 0;
    INTERNAL_SERVER_ERROR = 1;
  }
  Reason reason = 2;
}

message MessageAndCommand
{

enum  Command{
  CREATE_ACCOUNT_WEB =0;
CREATE_ACCOUNT_WEB_SUCCESS=1;
CREATE_ACCOUNT_WEB_FAILURE=2;
AUTHENTICATE_WEB=3;
AUTHENTICATE_WEB_SUCCESS=4;
AUTHENTICATE_WEB_FAILURE=5;
  }
  Command command =1;
string messageData=2;
}

